apps/launch/app.js

```diff
@@ -81,8 +84,14 @@
         // cleanup the timeout to not leave anything behind after being removed from ram
         if (lockTimeout) clearTimeout(lockTimeout);
         Bangle.removeListener("lock", lockHandler);
+        // Restore widgets if they were hidden by fullscreen setting
+        if (global.WIDGETS) require("widget_utils").show();
       }
     });
+    if (settings.fullscreen) {
+      require("widget_utils").hide();
```

Comment:
@bobrippling this makes me think you may be right regarding [L88](https://github.com/espruino/BangleApps/pull/3788/files#r2022414408)

Reply:
But we also have this consideration about maybe changing the back-function behaviour: https://github.com/espruino/BangleApps/pull/3788#issuecomment-2765635283

Reply:
That's true, it is a bit of a special case I suppose. Perhaps we try out a few different approaches?

Reply:
Ok - so in this case the widgets would have been hidden before (if there were any), but because we're using `back` a widget is being added anyway and so we're having to hide them again?

I feel we should probably just fix the `back` issue rather than having these lines in there?